revised divide by zero rule (#232)
rustfoundation/safety-critical-rust-coding-guidelines: src/coding-guidelines/expressions.rst

      :id: compl_ex_k1CD6xoZxhXb
      :status: draft

      Checked division prevents division by zero from occuring.
      The programmer can then handle the returned :std:`std::option::Option`.
      Using checked division and remainder is particularly important in the signed integer case,
      where arithmetic overflow can also occur when dividing the minimum representable value by -1.

      .. code-block:: rust

         // Using the checked division API
         let y = match 5i32.checked_div(0) {
             None => 0
             Some(r) => r
         };

         // Using the checked remainder API
         let z = match 5i32.checked_rem(0) {
             None => 0
             Some(r) => r
         };

   .. compliant_example::
      :id: compl_ex_k1CD6xoZxhXc
      :status: draft

>       This compliant solution creates a divisor using :std:`std::num::NonZero`,

Reviewer: ```suggestion
      This compliant solution creates a divisor using :std:`std::num::NonZero`.
```

A small typo

      :std:`std::num::NonZero` is a wrapper around primitive integer types that guarantees the contained value is never zero.
      :std:`std::num::NonZero` creates a new type that represents a value that is known not to be zero,

>       ensuring that functions operating on these types can safely assume that zero is not a possible input. 

Reviewer: ```suggestion
      :std:`std::num::NonZero::new` creates a new binding that represents a value that is known not to be zero.
      This in turn ensures that functions operating on its value can safely assume that they are not being given zero their input. 
```

Making the link more precise, and the wording as well. I've cut the composite sentence into its two components; I believe the paragraph is a bit more readable this way. Lemme know what you think @rcseacord

>       The test for arithmetic overflow in this compliant example is unnecessary because ``divisor`` is an unsigned integer type.

Reviewer: ```suggestion

      Note that the test for arithmetic overflow in this compliant example (which is a possibility for non-zero negative divisors, namely, ``-1``) is unnecessary because ``divisor`` is an unsigned integer type.
```

I'm putting this on a separate paragraph because it's a separate idea. Lemme know what you think, @rcseacord
